src/Microsoft.VisualStudio.ProjectSystem.Managed/ProjectSystem/UpToDate/BuildUpToDateCheck.Log.cs

```diff
@@ -213,6 +216,22 @@ public void UpToDate()
 
                 Info(nameof(Resources.FUTD_UpToDate));
             }
+
+            public readonly struct Scope : IDisposable
+            {
+                private readonly Log _log;
+
+                public Scope(Log log)
+                {
+                    _log = log;
+                    _log.Indent++;
+                }
+
+                public void Dispose()
+                {
+                    _log.Indent--;
+                }
+            }
```

Comment:
We used a similar design to this for our REST API generators. Hmmm, just looked over that codebase and it only exists [as an example now](https://github.com/Azure/autorest.csharp/blob/29cd031d5f89644e997d667181b2452318215fff/src/AutoRest.CSharp/LowLevel/Generation/LowLevelExampleComposer.cs#L650-L653). The bottom of the file contains [CodeScope](https://github.com/Azure/autorest.csharp/blob/29cd031d5f89644e997d667181b2452318215fff/src/AutoRest.CSharp/LowLevel/Generation/LowLevelExampleComposer.cs#L851-L883).